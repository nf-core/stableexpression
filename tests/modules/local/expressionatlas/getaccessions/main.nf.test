nextflow_process {

    name "Test Process EXPRESSIONATLAS_GETACCESSIONS"
    script "modules/local/expressionatlas/getaccessions/main.nf"
    process "EXPRESSIONATLAS_GETACCESSIONS"
    tag "getaccessions"
    tag "module"

    test("Solanum tuberosum one keyword") {

        tag "potato_two_kw"

        when {

            process {
                """
                input[0] = "solanum_tuberosum"
                input[1] = "potato"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test('Solanum tuberosum two keywords') {

        tag "potato_two_kw"

        when {

            process {
                """
                input[0] = "solanum_tuberosum"
                input[1] = "potato,phloem"
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

    test('Solanum tuberosum no keyword') {

        tag "potato_no_kw"

        when {

            process {
                """
                input[0] = "solanum_tuberosum"
                input[1] = ""
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
