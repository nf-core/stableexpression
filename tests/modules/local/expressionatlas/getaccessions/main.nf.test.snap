{
    "Solanum tuberosum no keyword": {
        "content": [
            {
                "0": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "1": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "nltk",
                        "3.9.1"
                    ]
                ],
                "txt": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-01T14:25:08.425684541"
    },
    "Solanum tuberosum one keyword": {
        "content": [
            {
                "0": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "1": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "nltk",
                        "3.9.1"
                    ]
                ],
                "txt": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-01T14:23:43.573176854"
    },
    "Solanum tuberosum two keywords": {
        "content": [
            {
                "0": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ],
                "1": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "python",
                        "3.12.8"
                    ]
                ],
                "2": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "requests",
                        "2.32.3"
                    ]
                ],
                "3": [
                    [
                        "EXPRESSIONATLAS_GETACCESSIONS",
                        "nltk",
                        "3.9.1"
                    ]
                ],
                "txt": [
                    "accessions.txt:md5,dad63ac7a1715277fa44567bc40b5872"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.3"
        },
        "timestamp": "2025-01-01T14:24:29.42124266"
    }
}